File: BookExamples/AppLinking/app/src/main/java/com/ebookfrenzy/applinking/MyDBHandler.java
package com.ebookfrenzy.applinking;

import com.ebookfrenzy.applinking.provider.MyContentProvider;

import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.content.Context;
import android.content.ContentValues;
import android.database.Cursor;
import android.content.ContentResolver;
import android.util.Log;

import java.util.ArrayList;
import java.util.List;

public class MyDBHandler extends SQLiteOpenHelper {

    private static final String TAG = "Database";
    private ContentResolver myCR;

    private static final int DATABASE_VERSION = 1;
    private static final String DATABASE_NAME = "landmarks.db";
    public static final String TABLE_LOCATIONS = "locations";

    public static final String COLUMN_ID = "_id";
    public static final String COLUMN_TITLE = "title";
    public static final String COLUMN_DESCRIPTION = "description";
    public static final String COLUMN_PERSONAL = "personal";

    private static final int id_col = 0;
    private static final int title_col = 1;
    private static final int desc_col = 2;
    private static final int personal_col = 3;

    public MyDBHandler(Context context, String name,
                       SQLiteDatabase.CursorFactory factory, int version) {
        super(context, DATABASE_NAME, factory, DATABASE_VERSION);
        myCR = context.getContentResolver();
    }

    @Override
    public void onCreate(SQLiteDatabase db) {

    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion,
                          int newVersion) {
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_LOCATIONS);
        onCreate(db);
    }

    public void addLandmark(Landmark landmark) {

        ContentValues values = new ContentValues();
        values.put(COLUMN_ID, landmark.getID());
        values.put(COLUMN_TITLE, landmark.getTitle());
        values.put(COLUMN_DESCRIPTION, landmark.getDescription());
        values.put(COLUMN_PERSONAL, landmark.getPersonal());

        myCR.insert(MyContentProvider.CONTENT_URI, values);

    }

    public ArrayList<Landmark> findAllLandmarks() {

        ArrayList<Landmark> landmarks = new ArrayList();


        String[] projection = {COLUMN_ID,
                COLUMN_TITLE, COLUMN_DESCRIPTION, COLUMN_PERSONAL };

        String selection = null;

        Cursor cursor = myCR.query(MyContentProvider.CONTENT_URI,
                projection, selection, null,
                null);

        if (cursor.moveToFirst()) {
            while (!cursor.isAfterLast()) {

                Landmark landmark = new Landmark();

                landmark.setID(cursor.getString(id_col));
                landmark.setTitle(cursor.getString(title_col));
                landmark.setDescription(cursor.getString(desc_col));
                landmark.setPersonal(cursor.getInt(personal_col));
                landmarks.add(landmark);
                cursor.moveToNext();

            }
        }
        //cursor.close();
        return landmarks;
    }

    public Landmark findLandmark(String landmarkId) {
        String[] projection = {COLUMN_ID,
                COLUMN_TITLE, COLUMN_DESCRIPTION, COLUMN_PERSONAL };

        String selection = "_id = \"" + landmarkId + "\"";

        Log.i(TAG, "SELECTION = " + selection);

        Cursor cursor = myCR.query(MyContentProvider.CONTENT_URI,
                projection, selection, null,
                null);

        Landmark landmark = new Landmark();

        if (cursor.moveToFirst()) {
            cursor.moveToFirst(); // IS this needed?
            landmark.setID(cursor.getString(0));
            landmark.setTitle(cursor.getString(1));
            landmark.setDescription(cursor.getString(2));
            landmark.setPersonal(cursor.getInt(3));
            cursor.close();
        } else {
            landmark = null;
        }
        return landmark;

    }

    public boolean deleteLandmark(String landmarkId) {

        boolean result = false;

        String selection = "_id = \"" + landmarkId + "\"";

        int rowsDeleted = myCR.delete(MyContentProvider.CONTENT_URI,
                selection, null);

        if (rowsDeleted > 0)
            result = true;

        return result;
    }

}

File: KotlinIguanaSamples/SQLDemo_provider/app/src/main/java/com/ebookfrenzy/sqldemo/MyDBHandler.kt
package com.ebookfrenzy.sqldemo

import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.content.Context
import android.content.ContentValues
import android.content.ContentResolver

import com.ebookfrenzy.sqldemo.provider.MyContentProvider

class MyDBHandler(context: Context, name: String?,
                  factory: SQLiteDatabase.CursorFactory?, version: Int) :
    SQLiteOpenHelper(context, DATABASE_NAME, factory, DATABASE_VERSION) {

    private val myCR: ContentResolver

    init {
        myCR = context.contentResolver
    }

    override fun onCreate(db: SQLiteDatabase) {
        val CREATE_CUSTOMERS_TABLE = ("CREATE TABLE " +
                TABLE_CUSTOMERS + "("
                + COLUMN_ID + " INTEGER PRIMARY KEY," +
                COLUMN_CUSTOMERNAME
                + " TEXT," + COLUMN_CUSTOMERPHONE + " TEXT" + ")")

        db.execSQL(CREATE_CUSTOMERS_TABLE)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int,
                           newVersion: Int) {
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_CUSTOMERS)
        onCreate(db)
    }

    fun addCustomer(customer: Customer) {
        val values = ContentValues()
        values.put(COLUMN_CUSTOMERNAME, customer.customerName)
        values.put(COLUMN_CUSTOMERPHONE, customer.customerPhone)

        myCR.insert(MyContentProvider.CONTENT_URI, values)
    }

    fun findCustomer(customername: String): Customer? {
        val projection = arrayOf(COLUMN_ID, COLUMN_CUSTOMERNAME,
            COLUMN_CUSTOMERPHONE)
        val selection = "customername = \"" + customername + "\""

        val cursor = myCR.query(MyContentProvider.CONTENT_URI,
            projection, selection, null, null)
        var customer: Customer? = null
        if (cursor != null) {
            if (cursor.moveToFirst()) {
                cursor.moveToFirst()
                val id = Integer.parseInt(cursor.getString(0))
                val customerName = cursor.getString(1)
                val customerPhone = cursor.getString(2)
                customer = Customer(id, customername, customerPhone)
                cursor.close()
            }
        }
        return customer
    }

    fun deleteCustomer(customername: String): Boolean {
        var result = false
        val selection = "customername = \"" + customername + "\""
        val rowsDeleted = myCR.delete(MyContentProvider.CONTENT_URI,
            selection, null)

        if (rowsDeleted > 0)
            result = true
        return result
    }

    companion object {
        private val DATABASE_VERSION = 1
        private val DATABASE_NAME = "customerDB.db"
        val TABLE_CUSTOMERS = "customers"
        val COLUMN_ID = "_id"
        val COLUMN_CUSTOMERNAME = "customername"
        val COLUMN_CUSTOMERPHONE = "customerphone"
    }

}